spice_agent/inference/actions.py

```diff
@@ -210,23 +378,26 @@ def run_pipeline(
                             variant=variant,
                             use_safetensors=True,
                         )
-
                         refiner = refiner.to(self.device)
 
-                        latents = pipe(
-                            prompt=text_input,
-                            output_type="latent",
-                            **stable_diffusion_options,
-                        ).images  # type: ignore
+                        # Generates additional embeddings for refiner
+                        prompt_embeddings_for_refiner = CompelEmbeddingsProvider(
+                            pipeline=refiner,
+                            device=self.device,
+                            prompt=prompt,
+                            negative_prompt=negative_prompt,
+                        )()
```

Comment:
not my favorite to see a call get instantiated then immediately called

"what" got called is not clear